File: Python/0938-e-range-sum-of-bst.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def rangeSumBST(self, root: TreeNode, L: int, R: int) -> int:
        if not root or L > R:
            return None
        self.sum = 0
        self.inorder(root, L, R)
        return self.sum
    
    def inorder(self, node, L, R):
        if not node:
            return
        if L <= node.val <= R:
            self.sum += node.val
        if L < node.val:
            self.inorder(node.left, L, R)
        if node.val < R:
            self.inorder(node.right, L, R)

File: leetcode/C++/0938-e-range-sum-of-bst.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int rangeSumBST(TreeNode* root, int L, int R) {
        if (!root) return 0;
        int res = 0;
        helper(root, res, L, R);
        return res;
    }
    
    void helper(TreeNode* root, int& res, int L, int R) {
        if (!root) return;
        if (root->val >= L && root->val <= R)
            res += root->val;
        if (root->val > L)
            helper(root->left, res, L, R);
        if (root->val < R)
            helper(root->right, res, L, R);
    }
};